src/test/java/org/hyperledger/fabric/gateway/impl/GatewayTest.java

```diff
@@ -82,6 +82,18 @@ public void testCloseGatewayClosesNetworks() {
         assertThat(channel.isShutdown()).isTrue();
     }
 
+    @Test
+    public void testCloseGatewayNotForceClosesNetworks() {
+        Gateway gateway = builder.forceClose(false).connect();
+        Channel channel = gateway.getNetwork("assumed").getChannel();
+
+        assertThat(((GatewayImpl)gateway).isForceClose()).isFalse();
+
+        gateway.close();
+
+        assertThat(channel.isShutdown()).isTrue();
+    }
```

Comment:
This test is essentially the same as the `testCloseGatewayClosesNetworks` test above it. It is useful to confirm that the underlying channel is closed but we are not testing the new behaviour that changes the way the close is called.

I think it would be worth re-implementing this test (and the one above) so they explicitly test the close behaviour, something like this:

```java
    @Test
    public void testCloseGatewayForceClosesNetworks() {
        HFClient mockClient = testUtils.newMockClient();
        Channel mockChannel = testUtils.newMockChannel("CHANNEL");
        Mockito.when(mockClient.getChannel("CHANNEL")).thenReturn(mockChannel);

        builder.client(mockClient);

        try (Gateway gateway = builder.connect()) {
            gateway.getNetwork("CHANNEL");
        }

        Mockito.verify(mockChannel).shutdown(true);
    }

    @Test
    public void testCloseGatewayWithForceCloseDisabledClosesNetworks() {
        HFClient mockClient = testUtils.newMockClient();
        Channel mockChannel = testUtils.newMockChannel("CHANNEL");
        Mockito.when(mockClient.getChannel("CHANNEL")).thenReturn(mockChannel);

        builder.client(mockClient);
        builder.forceClose(false);

        try (Gateway gateway = builder.connect()) {
            gateway.getNetwork("CHANNEL");
        }

        Mockito.verify(mockChannel).shutdown(false);
    }
```

Reply:
Thank you for telling us about it.
I have changed the code to explicitly test the close behaviour.